connect-inject/handler_test.go

```diff
@@ -640,7 +641,305 @@ func TestHandlerPortValue(t *testing.T) {
 				return
 			}
 
-			require.Equal(port, tt.Expected)
+			require.Equal(tt.Expected, port)
+		})
+	}
+}
+
+// Test consulNamespace function
+func TestConsulNamespace(t *testing.T) {
+	cases := []struct {
+		Name                       string
+		EnableNamespaces           bool
+		ConsulDestinationNamespace string
+		EnableK8SNSMirroring       bool
+		K8SNSMirroringPrefix       string
+		K8sNamespace               string
+		Expected                   string
+	}{
+		{
+			"namespaces disabled",
+			false,
+			"default",
+			false,
+			"",
+			"namespace",
+			"",
+		},
+
+		{
+			"namespaces disabled, mirroring enabled",
+			false,
+			"default",
+			true,
+			"",
+			"namespace",
+			"",
+		},
+
+		{
+			"namespaces disabled, mirroring enabled, prefix defined",
+			false,
+			"default",
+			true,
+			"test-",
+			"namespace",
+			"",
+		},
+
+		{
+			"namespaces enabled, mirroring diabled",
+			true,
+			"default",
+			false,
+			"",
+			"namespace",
+			"default",
+		},
+
+		{
+			"namespaces enabled, mirroring disabled, prefix defined",
+			true,
+			"default",
+			false,
+			"test-",
+			"namespace",
+			"default",
+		},
+
+		{
+			"namespaces enabled, mirroring enabled",
+			true,
+			"default",
+			true,
+			"",
+			"namespace",
+			"namespace",
+		},
+
+		{
+			"namespaces enabled, mirroring enabled, prefix defined",
+			true,
+			"default",
+			true,
+			"test-",
+			"namespace",
+			"test-namespace",
+		},
+	}
+
+	for _, tt := range cases {
+		t.Run(tt.Name, func(t *testing.T) {
+			require := require.New(t)
+
+			h := Handler{
+				EnableNamespaces:           tt.EnableNamespaces,
+				ConsulDestinationNamespace: tt.ConsulDestinationNamespace,
+				EnableK8SNSMirroring:       tt.EnableK8SNSMirroring,
+				K8SNSMirroringPrefix:       tt.K8SNSMirroringPrefix,
+			}
+
+			ns := h.consulNamespace(tt.K8sNamespace)
+
+			require.Equal(tt.Expected, ns)
+		})
+	}
+}
+
+// Test shouldInject function
+func TestShouldInject(t *testing.T) {
+	cases := []struct {
+		Name                  string
+		Pod                   *corev1.Pod
+		K8sNamespace          string
+		EnableNamespaces      bool
+		AllowK8sNamespacesSet mapset.Set
+		DenyK8sNamespacesSet  mapset.Set
+		Expected              bool
+	}{
+		{
+			"kube-system not injected",
+			&corev1.Pod{
+				ObjectMeta: metav1.ObjectMeta{
+					Annotations: map[string]string{
+						// Service annotation is required for injection
+						annotationService: "testing",
+					},
+				},
+			},
+			"kube-system",
+			false,
+			mapset.NewSet(),
+			mapset.NewSet(),
+			false,
+		},
+		{
+			"kube-public not injected",
+			&corev1.Pod{
+				ObjectMeta: metav1.ObjectMeta{
+					Annotations: map[string]string{
+						annotationService: "testing",
+					},
+				},
+			},
+			"kube-public",
+			false,
+			mapset.NewSet(),
+			mapset.NewSet(),
+			false,
+		},
+		{
```

Comment:
The important part of the tests are whether the namespace we're coming from is contained in either the allow or deny list and if that gives us the correct response about whether we should inject. These tests would cover the same cases even if I used a namespace other than default.

Reply:
What I was trying to say was that using only one namespace is insufficient. For example, for the test case I mentioned there, asserting only on `shouldInject(..., default) == false` is not fully covering that case because it doesn't check that the namespace in the allow list is still allowed. We should be checking both the allow and deny behaviors, but using a single namespace in assertions only allows us to check one. Hope this makes sense 😀